test/Microsoft.Framework.DependencyInjection.Tests/ActivatorUtilitiesTests.cs

```diff
@@ -79,12 +79,24 @@ public void TypeActivatorWorksWithStaticCtor(CreateInstanceFunc createFunc)
             Assert.NotNull(anotherClass);
         }
 
-        [Theory]
-        [MemberData(nameof(CreateInstanceFuncs))]
-        public void TypeActivatorWorksWithCtorWithOptionalArgs(CreateInstanceFunc createFunc)
```

Comment:
This is not the way CreateInstance behaves.

You could fix the behavior by calling GetService instead of GetDefaultService iff `constructorParameters[i].HasDefaultValue` is true.